client/state/jetpack-connect/reducer.js

```diff
@@ -182,10 +188,12 @@ export function jetpackConnectAuthorize( state = {}, action ) {
 			);
 		case JETPACK_CONNECT_QUERY_SET:
 			const queryObject = Object.assign( {}, action.queryObject );
+			const shouldAutoAuthorize = ( 'woocommerce-services-auto-authorize' === get( queryObject, 'from' ) );
 			return Object.assign(
 				{},
 				buildDefaultAuthorizeState(),
-				{ queryObject: queryObject }
+				{ queryObject: queryObject },
+				{ autoAuthorize: shouldAutoAuthorize }
```

Comment:
Any reason to have `autoAuthorize: shouldAutoAuthorize` in a separate, new object, rather than in the previous one that we're assigning from?